packages/blocks/src/api/serializer.js

```diff
@@ -219,7 +219,7 @@ export function getCommentAttributes( blockType, attributes ) {
 			// Ignore default value.
 			if (
 				'default' in attributeSchema &&
-				attributeSchema.default === value
+				isEqual( attributeSchema.default, value )
```

Comment:
isEqual is avoided for performance reasons I believe. How are you "resetting" the default value, you could be using  the same instance?

That said, if there's no measurable impact, I'm fine with this change personally.

Reply:
I know that there might be performance "issues" when using isEqual but in that specific case, I don't expect any significant ones because it will just compare the value with a default one which is not expected to be that complex. So IMO there will be no measurable impart.

Regarding your other question, I'm sorry, but I'm not really following.

Reply:
How are you empetting the value in your block. Basically you should be able to do something like `setAttributes( { myAttribute: myBlocksSettings.attributes.myAttribute.default } )` to reset the value and don't suffer from this issue.

Reply:
OK, now I get it. To reset the value in my block I had to do something like `setAttributes( { myAttribute: {} } )`, since the default value for my attribute was an empty object, and it was enough for my specific case.

Before this change, even though my attribute value matched the default value, it was always saved in the post content no matter what.

Reply:
I'm not against merging this to make it simpler for block authors but it looks like there's some test failures?

Reply:
For the record, in order to solve my issue without this change, I had to do `setAttributes( { myAttribute: undefined } )`. I could have never thought of this if I hadn't checked the tests for this method. :)

Reply:
I'm not really sure that the test failures you are mentioning are because of my change. Is there any way that I can check?

Reply:
there's a linting issue, a unit tests that is most likely because of the change, the e2e test need confirmation but if you  rerun the check again and it fails, it means it's because  of the change here.

Reply:
You were right but it should be good to go now. The failing e2e tests here are the same that fail on the checks for `origin/master` https://github.com/WordPress/gutenberg/runs/1537564269.

Reply:
You were right but it should be good to go now. The e2e tests are failing randomly. I don't know what to do there.

Reply:
All checks are green @youknowriad 😄

Reply:
I feel reluctant about this change. Switching from strict equality to any kind of rich comparison seems like a dangerous precedent to set, muddying attribute semantics and introducing more unexpected situations than it fixes.

Yes, performance is one concern, but even if there were no measurable performance impact from this change (for which we can't just rely on the performance tests, since this is not the kind of operation that they target) I would rather not merge this.

I concede that "resetting" an attribute of type `'object'` can be confusing for block developers, but it's also a direct consequence of the flexibility and strangeness of object types in JS. But, as much as we like to mock [JS's type relationships](https://user-images.githubusercontent.com/150562/102260172-fa348900-3f07-11eb-944f-3bc6a1200281.png), I think JS is right about the semantics of objects in one regard: an empty object `{}` _should_ be different from "no value" or "default value". Developers should retain the ability to set an attribute's value to an explicit `{}` value without the framework cleverly coercing that value into nothingness.

Thus, the practice of unsetting an attribute explicitly by assigning `schema[attributeName].default` or `undefined` to it should remain the standard, and my recommendation is to improve the documentation to reflect this.

Reply:
Understood and fine for me. I've found a way to do my job anyway. Thanks for the review.

Reply:
Thanks for the proposal, regardless!